nursery_store.go

```diff
@@ -0,0 +1,1377 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/boltdb/bolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/wire"
+)
+
+// NurseryStore abstracts the persistent storage layer for the utxo nursery.
+// Concretely, it stores commitment and htlc outputs that until any time-bounded
+// constraints have fully matured. The store exposes methods for enumerating
+// its contents, and persisting state transitions detected by the utxo nursery.
+type NurseryStore interface {
+
+	// Incubate registers a commitment output and a variadic number of htlc
+	// outputs to be swept back into the user's wallet. The event is
+	// persisted to disk, such that the nursery can resume the incubation
+	// process after a potential crash.
+	Incubate(*kidOutput, ...babyOutput) error
```

Comment:
Any reason `babyOutput` is variadic instead of an array?

Reply:
Not really, originally I had some use cases in mind where the second argument was unnecessary, so this is mostly an artifact of that. I do agree though that it makes more sense as a slice, given how we will use it in the utxn